ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,321 @@
+module ManoptJuMPExt
+
+using LinearAlgebra
+import JuMP
+const MOI = JuMP.MOI
+using ManifoldsBase
+using ManifoldDiff
+
+function __init__()
+    # So that the user can use the convenient `Manopt.Optimizer`
+    setglobal!(Manopt, :Optimizer, Optimizer)
+end
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+function MOI.dimension(set::VectorizedManifold)
+    return prod(ManifoldsBase.representation_size(set.manifold))
+end
+
+"""
+    Manopt.Optimizer()
+
+Creates a new optimizer object.
+
+The minimization of a function `f(X)` of an an array `X[1:n1,1:n2,...]`
+over a manifold `M` starting at `X0`, can be modeled as follows:
+```julia
+using JuMP
+@variable(model, X[i1=1:n1,i2=1:n2,...] in M, start = X0[i1,i2,...])
+@objective(model, Min, f(X))
+```
+"""
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    manifold::Union{Nothing,ManifoldsBase.AbstractManifold}
+    problem::Union{Nothing,AbstractManoptProblem}
+    state::Union{Nothing,AbstractManoptSolverState}
+    variable_primal_start::Vector{Union{Nothing,Float64}}
+    sense::MOI.OptimizationSense
+    nlp_model::MOI.Nonlinear.Model
+    options::Dict{String,Any}
+    function Optimizer()
+        return new(
+            nothing,
+            nothing,
+            nothing,
+            Union{Nothing,Float64}[],
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
```

Comment:
```suggestion
```

Reply:
If you check that the sense is feasibility, it doesn't matter if there is an objective here.

Reply:
I guess so but it's kind of weird. If the user sets a sense then the objective is going to appear. I feel it would be easier if `nlp_model` where implementing the `empty`-part of the API. The sub-models for variables, constraints and objective defined in `MOI.Utilities` implement the MOI interface even if they only have one part of the model and it works fine.

Reply:
Was this done? Then we could close it.

Reply:
Yes, the necessary feature has been added in https://github.com/jump-dev/MathOptInterface.jl/pull/2305 so it's fixed now